js/tree.js

```diff
@@ -372,11 +503,36 @@
 	};
 
 	$.fn.tree.defaults = {
-		dataSource: function (options, callback) {},
+		dataSource: function dataSource(options, callback) {},
 		multiSelect: false,
 		cacheItems: true,
 		folderSelect: true,
-		itemSelect: true
+		itemSelect: true,
+		/*
+		* currently calling "open" again on a folder will close it. Setting
+		* `ignoreRedundantOpens` to `true` will make the folder instead ignore
+		* the redundant call and stay open.
+		*/
+		ignoreRedundantOpens: false,
```

Comment:
So that this isn't a breaking change. Calling "open" on something, should do that. However, the current API instead treats "open" as a "toggle" and will close a folder that is open if you call `openFolder` on it. This switch allows you to fix the API until 3.7.x when we can deprecate the switch and make `openFolder` behave correctly by default. One of the first problems I encountered when trying to build the `discloseAll` function was that suddenly folders were closing themselves when I was accidentally calling `openFolder` on already opened folders. This was annoying, confusing, and hard to debug.